### PR TITLE
Retry with testsuite

### PR DESCRIPTION
**Pull request by GavinWu**
Issue with original code:
- Was not able to get the Retry attribute to work with [TestCase()] which a lot of my tests uses to pass in parameters

Changes:
Took out the interFixture class as the tries/ test run variable was not being reset per test. So I would run one_out_of_three and then run the inherent fixture and the test would pass right away because the static class is still holding onto that test run from previous test. 

Using the private method within the fixture will allow me to execute a test at a time but without the need to rebuild the test.dll to reset the variable(run count)

Added the required pieces to work with Parameterized test suite
Made sure the attribute does not erase the category attribute of the test/Test suite
Added .Inconclusive as one of the failing scenario for TestResult as we use that if the test fails at setup(place where we create selenium browser)

The Parameterized test suite changes was partially taken from  http://stackoverflow.com/questions/14053948/how-to-write-an-add-in-to-extend-individual-tests-within-nunit-testcases-in-para

### CODE REVIEW

File: src/NUnit-retry/RetriedTestSuiteMethod.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            _tryCount = run;
+            _requiredPassCount = requiredPass;
+        }
+
+        public override TestResult Run(EventListener listener, ITestFilter filter)
+        {
+            var successCount = 0;
+            TestResult failureResult = null;
+
+            for (var i = 0; i < _tryCount; i++)
```
Comment (GavinWu) [inline src/NUnit-retry/RetriedTestSuiteMethod.cs:29]: thought about taking this loop logic out and make a sharable method for this class and RetriedTestMEthod, but the loop uses a lot of variables (5, event, filter, trycount, run, base) so decided to duplicate the code

File: src/NUnit-retry/RetryAttribute.cs [hunk 2]
```diff
@@ -11,15 +11,15 @@ namespace NUnit_retry
     [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
     public class RetryAttribute : Attribute
     {
-        public RetryAttribute(int times = 5, int requiredPassCount = 1)
+        public RetryAttribute(int times = 3, int requiredPassCount = 1)
```
Comment (GavinWu) [inline src/NUnit-retry/RetryAttribute.cs:14]: Personal preference as the Retry will be used on Selenium tests which runs 1-6minute and 5 tries would take up too much resources

File: src/NUnit-retry.Tests/MethodTests.cs [hunk 3]
```diff
@@ -11,38 +13,138 @@ namespace NUnit_retry.Tests
...
+            Console.WriteLine("Passed");
+            _run++;
+
+            Assert.Pass();
         }
 
-        [Test]
-        [Retry]
-        public void When_SucceedOnce_Pass()
+        [TestCase(0, "abc"), Category("E")]
```
Comment (GavinWu) [inline src/NUnit-retry.Tests/MethodTests.cs:117]: The random Category and TestCase arguments are for testing purposes. Just need to make sure Category and arguments are properly preserved

**Comment by slvnperron** [comment]
Thanks @GavinWu for the contribution. I pulled the work and put it in branch PR2.
When modifying the tests, you did break the test cases. You can't share the _run variable like you did.. all the tests worked on the same instance and all your tests were wrong.

Please have a look at the branch PR2. I fixed the tests for you, but unfortunately I can't get your contribution to work. The tests all fail (the Retry attribute just doesn't work anymore)

Here's some guidelines that you need to follow when writing tests:
- Local variables are shared across all tests. You must use the TestsHelper I wrote
- Test names should be unique
- The TestsHelper is based solely on the calling method name. It doesn't care about the category or the parameters

Please fix the library then do your pull request against PR2 again.

**Comment by slvnperron** [comment]
Also check out the new Master branch. I merged contributions from @fwitt. It supports parametrization and it works.. there's just the TestCase and TestCaseSource that needs to be fixed.

**Comment by fwitt** [comment]
TestCase and TestCaseSource should work with my changes if you specify the Retry attribute at the method level rather than the class level. I meant to come back and look at making both work but I had it working for the scenario I needed so it got bumped down my priority order.

**Comment by slvnperron** [comment]
Alright will test that in a few hours. Thanks @fwitt

**Comment by slvnperron** [comment]
That worked. Thanks

**Pull request closed without merging**